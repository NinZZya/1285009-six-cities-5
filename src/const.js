export const AppPath = {
  ROOT: `/`,
  LOGIN: `/login`,
  CITY: `/city`,
  OFFER: `/offer`,
  FAVORITES: `/favorites`,
  NOT_FOUND: `/not-found`,
};

export const CITIES = {
  1: {id: 1, name: `Paris`},
  2: {id: 2, name: `Cologne`},
  3: {id: 3, name: `Brussels`},
  4: {id: 4, name: `Amsterdam`},
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};

export const SortType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const STARS = [
  {value: 5, title: `perfect`},
  {value: 4, title: `good`},
  {value: 3, title: `not bad`},
  {value: 2, title: `badly`},
  {value: 1, title: `terribly`},
];
