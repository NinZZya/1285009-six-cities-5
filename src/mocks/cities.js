export default {
  1: {
    id: 1,
    name: `Paris`,
    coords: [48.864716, 2.349014],
  },
  2: {
    id: 2,
    name: `Cologne`,
    coords: [50.9415491, 6.9408247],
  },
  3: {
    id: 3,
    name:
    `Brussels`,
    coords: [50.8503396, 4.3517103],
  },
  4: {
    id: 4,
    name: `Amsterdam`,
    coords: [52.38333, 4.9],
  },
  5: {id: 5, name: `Hamburg`},
  6: {id: 6, name: `Dusseldorf`},
};
