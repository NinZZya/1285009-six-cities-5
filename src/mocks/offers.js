export default [
  {
    id: 0,
    city: {
      id: 1,
      name: `Paris`,
    },
    title: `Wood and stone place`,
    type: `Private room`,
    price: 228,
    rate: 0,
    bedroomsCount: 1,
    adultsCount: 1,
    features: [
      `Dishwasher`,
      `Baby seat`,
    ],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isPro: false,
    },
    images: [
      `/img/apartment-03.jpg`,
      `/img/apartment-02.jpg`,
    ],
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    reviews: [
      {
        id: 0,
        user: {
          id: 3,
          name: `John`,
          avatar: `/img/avatar.svg`,
        },
        date: `2019-11-26`,
        rate: 2,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      },
      {
        id: 1,
        user: {
          id: 3,
          name: `John`,
          avatar: `/img/avatar.svg`
        },
        date: `2019-03-23`,
        rate: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      }
    ],
    isPremium: false,
  },
  {
    id: 1,
    city: {
      id: 1,
      name: `Paris`,
    },
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    price: 97,
    rate: 3,
    bedroomsCount: 1,
    adultsCount: 3,
    features: [
      `Dishwasher`,
      `Towels`,
      `Fridge`,
      `Heating`,
      `Washing machine`,
      `Wi-Fi`,
      `Coffee machine`,
    ],
    host: {
      name: `John`,
      avatar: `/img/avatar.svg`,
      isPro: false,
    },
    images: [
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
    ],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    reviews: [
      {
        id: 2,
        user: {
          id: 4,
          name: `Ann`,
          avatar: `/img/avatar.svg`
        },
        date: `2018-09-22`,
        rate: 2,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      },
      {
        id: 3,
        user: {
          id: 1,
          name: `Max`,
          avatar: `/img/avatar-max.jpg`
        },
        date: `2018-10-24`,
        rate: 0,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      },
      {
        id: 4,
        user: {
          id: 4,
          name: `Ann`,
          avatar: `/img/avatar.svg`
        },
        date: `2019-08-16`,
        rate: 1,
        text: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      }
    ],
    isPremium: false,
  },
  {
    id: 2,
    city: {
      id: 2,
      name: `Paris`
    },
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    price: 89,
    rate: 1,
    bedroomsCount: 2,
    adultsCount: 1,
    features: [],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isPro: false,
    },
    images: [
      `/img/apartment-01.jpg`
    ],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    reviews: [
      {
        id: 5,
        user: {
          id: 4,
          name: `Ann`,
          avatar: `/img/avatar.svg`,
        },
        date: `2019-06-07`,
        rate: 3,
        text: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      },
      {
        id: 6,
        user: {
          id: 1,
          name: `Max`,
          avatar: `/img/avatar-max.jpg`,
        },
        date: `2019-06-02`,
        rate: 1,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      }
    ],
    isPremium: false,
  },
  {
    id: 3,
    city: {
      id: 2,
      name: `Paris`,
    },
    title: `Wood and stone place`,
    type: `Private room`,
    price: 381,
    rate: 3,
    bedroomsCount: 1,
    adultsCount: 1,
    features: [
      `Fridge`,
      `Towels`,
    ],
    host: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      isPro: false
    },
    images: [
      `/img/room.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
    ],
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    reviews: [
      {
        id: 7,
        user: {
          id: 4,
          name: `Ann`,
          avatar: `/img/avatar.svg`,
        },
        date: `2018-11-08`,
        rate: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      },
      {
        id: 8,
        user: {
          id: 2,
          name: `Angelina`,
          avatar: `/img/avatar-angelina.jpg`
        },
        date: `2019-02-23`,
        rate: 0,
        text: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      }
    ],
    isPremium: false,
  }
];
