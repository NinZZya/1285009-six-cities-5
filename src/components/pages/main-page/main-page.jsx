import React from 'react';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';
import PageContainer from '@/components/page-container/page-container';
import Container from '@/components/container/container';
import CitiesTabs from '@/components/cities-tabs/cities-tabs';
import Sort from '@/components/sort/sort';
import OffersList, {OffersListType} from '@/components/offers-list/offers-list';
import NoOffers from '@/components/no-offers/no-offers';
import LoadingData from '@/components/loading-data/loading-data';
import Map from '@/components/map/map';
import withActiveId from '@/hocs/with-active-id/with-active-id';
import * as CitiesAction from '@/reducer/cities/cities-actions';
import * as CitiesSelector from '@/reducer/cities/cities-selectors';
import * as OffersAction from '@/reducer/offers/offers-actions';
import * as OffersSelector from '@/reducer/offers/offers-selectors';
import * as Type from '@/types';
import {
  SortType,
  AppPath,
  IdName,
  DataStatus,
  DEFAULT_CITY_ID,
} from '@/const';


const SORTS = Object.values(SortType);

const ContainerType = {
  PAGE: `index`,
  CITIES: `cities__places`,
};

const getCityId = (match) => {
  if (match.path !== AppPath.ROOT) {
    const cityId = Number(match.params[IdName.CITY]);
    return cityId && !isNaN(cityId) ? cityId : -1;
  }

  return DEFAULT_CITY_ID;
};

const getOffersContent = (args) => {
  const {
    activeId,
    onActiveIdChange,
    offers,
    cities,
    activeCityId,
    sortType,
    changeOffersSortType,
  } = args;

  const activeCity = cities[activeCityId];

  if (offers.length) {

    return (
      <Container type={ContainerType.CITIES}>
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">
            {`${offers.length} places to stay in ${activeCity.name}`}
          </b>
          <Sort
            activeSort={sortType}
            sorts={SORTS}
            onSortClick={(type) => {
              changeOffersSortType(type);
            }}
          />
          <OffersList
            type={OffersListType.MAIN}
            offers={offers}
            onOfferHover={onActiveIdChange}
          />
        </section>
        <div className="cities__right-section">
          <section className="cities__map  map">
            <Map
              center={activeCity}
              pins={offers}
              activeId={activeId}
            />
          </section>
        </div>
      </Container>
    );
  }

  return <NoOffers city={activeCity} />;
};

const MainPage = (props) => {
  const {
    activeCityId,
    offersStatus,
    offers,
    cities,
    chageActiveCityId,
    match,
  } = props;

  const pathCityId = getCityId(match);
  const pathCity = cities[pathCityId];

  if (!pathCity || pathCityId === -1) {
    return <Redirect to={AppPath.NOT_FOUND} />;
  }

  if ((pathCityId !== activeCityId) && pathCity) {
    chageActiveCityId(pathCityId);
  }

  const offersContent = offersStatus === DataStatus.SUCCESS ?
    getOffersContent(props) :
    null;

  const isEmpty = !offers.length;

  return (
    <PageContainer
      type={ContainerType.PAGE}
      empty={isEmpty}
    >
      <h1 className="visually-hidden">Cities</h1>
      <CitiesTabs
        activeCityId={activeCityId}
        cities={cities}
      />
      <div className="cities">
        <Container
          type={ContainerType.CITIES}
          empty={isEmpty}
        >
          <LoadingData status={offersStatus} />
          {offersContent}
        </Container>
      </div>
    </PageContainer>
  );
};

MainPage.propTypes = {
  activeId: Type.ID,
  onActiveIdChange: Type.FUNCTION,
  offersStatus: Type.DATA_STATUS,
  offers: Type.LIST_OFFERS,
  cities: Type.CITIES,
  activeCityId: Type.ID,
  sortType: Type.SORT,
  chageActiveCityId: Type.FUNCTION,
  changeOffersSortType: Type.FUNCTION,
  match: Type.MATCH,
};

const mapStateToProps = (state) => ({
  offersStatus: OffersSelector.getOffersStatus(state),
  offers: OffersSelector.getSortedCityOffers(state),
  cities: CitiesSelector.getCities(state),
  activeCityId: CitiesSelector.getActiveCityId(state),
  sortType: OffersSelector.getOffersSortType(state),
});

const mapDispatchToProps = (dispatch) => ({
  chageActiveCityId: (cityId) => {
    dispatch(CitiesAction.changeActiveCityId(cityId));
  },
  changeOffersSortType: (sortType) => {
    dispatch(OffersAction.changeOffersSortType(sortType));
  },
});


export {MainPage};
export default withActiveId(
    connect(mapStateToProps, mapDispatchToProps)(MainPage)
);
