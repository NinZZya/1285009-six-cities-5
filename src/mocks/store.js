import NameSpace from '../store/reducer/name-space';
import offers, {offersMap} from './offers';
import cities from './cities';
import reviews from './reviews';
import {user} from './users';
import {SortType, DataStatus, UserStatus} from '../constants/const';

const ACTIVE_CITY_ID = 4;
const ERROR = `ERROR`;
const DATA_STATUS = DataStatus.SUCCESS;
const SORT_TYPE = SortType.HIGH_TO_LOW;

const citiesState = {
  cities,
  activeId: ACTIVE_CITY_ID,
};

const offersState = {
  status: DATA_STATUS,
  offers: offersMap,
  favorites: offersMap,
  sortType: SORT_TYPE,
  offerStatus: DATA_STATUS,
  reviewsStatus: DATA_STATUS,
  reviews,
  nearOffersStatus: DATA_STATUS,
  nearOffers: offers,
};

const userState = {
  status: UserStatus.NO_AUTH,
  user,
  error: ERROR,
};

export default {
  [NameSpace.CITIES]: citiesState,
  [NameSpace.OFFERS]: offersState,
  [NameSpace.USER]: userState,
};
