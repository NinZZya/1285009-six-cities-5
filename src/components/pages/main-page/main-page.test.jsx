import React from 'react';
import {Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {MainPage} from './main-page';
import store from '../../../store/store';
import history from '../../../constants/history';
import {DataStatus, SortType} from '../../../constants/const';
import offers from '../../../mocks/offers';
import cities from '../../../mocks/cities';


const ACTIVE_CITY_ID = 4;
const testing = () => {};

describe(`Should MainPage render correctly`, () => {
  it(`Should MainPage render correctly loading`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <MainPage
              offersStatus={DataStatus.LOADING}
              cities={cities}
              activeCityId={ACTIVE_CITY_ID}
              sortType={SortType.POPULAR}
              onChageActiveCityId={testing}
              onChangeOffersSortType={testing}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should MainPage render correctly error`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <MainPage
              offersStatus={DataStatus.LOADING}
              cities={cities}
              activeCityId={ACTIVE_CITY_ID}
              sortType={SortType.POPULAR}
              onChageActiveCityId={testing}
              onChangeOffersSortType={testing}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should MainPage render correctly success with offers`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <MainPage
                offersStatus={DataStatus.SUCCESS}
                offers={offers}
                cities={cities}
                activeCityId={ACTIVE_CITY_ID}
                sortType={SortType.POPULAR}
                onChageActiveCityId={testing}
                onChangeOffersSortType={testing}
              />
            </Router>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
