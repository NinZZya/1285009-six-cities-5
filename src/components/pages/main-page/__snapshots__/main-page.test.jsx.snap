// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainPage render correctly Should MainPage render correctly error 1`] = `
<main
  className="page__main page__main--index page__main--index-empty"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/1"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/2"
            onClick={[Function]}
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/3"
            onClick={[Function]}
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="/city/4"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/5"
            onClick={[Function]}
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/6"
            onClick={[Function]}
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="container cities__places-container cities__places-container--empty"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "38px",
            "fontStyle": "oblique",
            "fontWeight": 700,
            "justifyContent": "flex-start",
            "lineHeight": 0.7,
            "margin": "20px",
            "padding": "30px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <h3
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "space-between",
            }
          }
        >
          <img
            alt="logo"
            height="50"
            src="/img/logo.svg"
            style={
              Object {
                "marginRight": "20px",
              }
            }
            width="100"
          />
          <svg
            height="100%"
            version="1.1"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <path
              id="path"
            >
              <animate
                attributeName="d"
                begin="0s"
                dur="3s"
                from="m0,110 h0"
                repeatCount="indefinite"
                to="m0,110 h1100"
              />
            </path>
            <text
              fill="black"
            >
              <textPath
                xlinkHref="#path"
              >
                Loading...
              </textPath>
            </text>
          </svg>
        </h3>
      </div>
    </div>
  </div>
</main>
`;

exports[`Should MainPage render correctly Should MainPage render correctly loading 1`] = `
<main
  className="page__main page__main--index page__main--index-empty"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/1"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/2"
            onClick={[Function]}
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/3"
            onClick={[Function]}
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="/city/4"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/5"
            onClick={[Function]}
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/6"
            onClick={[Function]}
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="container cities__places-container cities__places-container--empty"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "fontSize": "38px",
            "fontStyle": "oblique",
            "fontWeight": 700,
            "justifyContent": "flex-start",
            "lineHeight": 0.7,
            "margin": "20px",
            "padding": "30px",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <h3
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "space-between",
            }
          }
        >
          <img
            alt="logo"
            height="50"
            src="/img/logo.svg"
            style={
              Object {
                "marginRight": "20px",
              }
            }
            width="100"
          />
          <svg
            height="100%"
            version="1.1"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <path
              id="path"
            >
              <animate
                attributeName="d"
                begin="0s"
                dur="3s"
                from="m0,110 h0"
                repeatCount="indefinite"
                to="m0,110 h1100"
              />
            </path>
            <text
              fill="black"
            >
              <textPath
                xlinkHref="#path"
              >
                Loading...
              </textPath>
            </text>
          </svg>
        </h3>
      </div>
    </div>
  </div>
</main>
`;

exports[`Should MainPage render correctly Should MainPage render correctly success with offers 1`] = `
<main
  className="page__main page__main--index "
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/1"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/2"
            onClick={[Function]}
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/3"
            onClick={[Function]}
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="/city/4"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/5"
            onClick={[Function]}
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="/city/6"
            onClick={[Function]}
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="container cities__places-container "
    >
      <div
        className="container cities__places-container "
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            5 places to stay in Amsterdam
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom "
            >
              <li
                className="places__option places__option--active"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option "
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option "
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option "
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="places__list cities__places-list tabs__content"
          >
            <article
              className="place-card cities__place-card"
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  className="place-card__image-link"
                  href="/offer/0"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                >
                  <img
                    alt="Image of Apartment. "
                    className="place-card__image"
                    height={200}
                    src="/img/apartment-01.jpg"
                    width={260}
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      120
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                       night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button "
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="rating__stars place-card__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    className="place-card__name-link"
                    href="/offer/0"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    Beautiful & luxurious apartment at great location
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
            <article
              className="place-card cities__place-card"
            >
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  className="place-card__image-link"
                  href="/offer/1"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                >
                  <img
                    alt="Image of Private room. "
                    className="place-card__image"
                    height={200}
                    src="/img/room.jpg"
                    width={260}
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      80
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                       night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button place-card__bookmark-button--active"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="rating__stars place-card__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    className="place-card__name-link"
                    href="/offer/1"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    Wood and stone place
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Private room
                </p>
              </div>
            </article>
            <article
              className="place-card cities__place-card"
            >
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  className="place-card__image-link"
                  href="/offer/2"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                >
                  <img
                    alt="Image of Apartment. "
                    className="place-card__image"
                    height={200}
                    src="/img/apartment-02.jpg"
                    width={260}
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      132
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                       night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button place-card__bookmark-button--active"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="rating__stars place-card__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    className="place-card__name-link"
                    href="/offer/2"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    Canal View Prinsengracht
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
            <article
              className="place-card cities__place-card"
            >
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  className="place-card__image-link"
                  href="/offer/3"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                >
                  <img
                    alt="Image of Apartment. "
                    className="place-card__image"
                    height={200}
                    src="/img/apartment-03.jpg"
                    width={260}
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      180
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                       night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button "
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="rating__stars place-card__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    className="place-card__name-link"
                    href="/offer/3"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    Nice, cozy, warm big bed apartment
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
            <article
              className="place-card cities__place-card"
            >
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  className="place-card__image-link"
                  href="/offer/4"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                >
                  <img
                    alt="Image of Apartment. "
                    className="place-card__image"
                    height={200}
                    src="/img/room.jpg"
                    width={260}
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      180
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                       night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button "
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="rating__stars place-card__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    className="place-card__name-link"
                    href="/offer/4"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    Wood and stone place
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map  map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </div>
</main>
`;
